### PR TITLE
feat: custom ci docs

### PR DESCRIPTION
**Pull request by xuzhu-591**

### CODE REVIEW

File: docs/tutorials/custom-ci.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+title: Custom CI
+toc_max_heading_level: 6
+---
+
+## How does Horizon CI work
+Horizon CI works based on [tekton pipeline](https://github.com/tektoncd/pipeline) and [tekton triggers](https://github.com/tektoncd/triggers). A tekton pipeline will be triggered for a incoming buildDeploy request to perform the typical steps, such as compiling and pushing image. Then, the pipeline output that holds artifact image url and other information will be returned to Horizon core by callback. Horizon CI pipeline includes two tasks that is shown below:
+![pipeline](/image/pipeline.png)
+* Git step: use token to pull user's source code.
+* Compile step: compile and package the source code, such as `mvn clean package`.
+* image step: build dockerfile to store the artifact and push the generated image to registry.
```
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:11]: Build dockerfile? That seems not making sense. Maybe `build image from dockerfile`?

File: docs/tutorials/custom-ci.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+* deploy step: assemble pipeline output and callback horizon api.
+
+## How to customize your CI
+**Custom CI is supported partly on Horizon now. In order to custom CI, you may prepare two parts of modification.**
```
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:15]: In order to customize CI

File: docs/tutorials/custom-ci.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+### Build config
+Build config holds user's build configurations of application or cluster, which is characterized as json object `pipelineJSONBlob` in pipeline.
+![](/image/new-app-demo2.png)
+You can customize own build config based on [json schema](https://json-schema.org), which contains two parts: json schema and ui schema. Override them in values.yaml of [horizon chart](https://github.com/horizoncd/helm-charts), which has the following structure:
```
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:20]: You can customize your own build. 
And build config -> building config? 
Link of Horizon chart -> https://github.com/horizoncd/helm-charts/tree/main/charts/horizon

File: docs/tutorials/custom-ci.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          "inline": true
+        }
+      }
+    }
+```
+You can refer to [json schema](https://json-schema.org) and [json schema in horizon template](/docs/reference/template/specification.md#json-schema) for more information.
+
+
+### Step scripts
+The step scripts in values.yaml also can be overrode to perform various pipelines according to user's build config above. The associated file structure is as follows:
```
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:75]: Add link on values.yaml with https://github.com/horizoncd/helm-charts/blob/main/charts/horizon/values.yaml

File: docs/tutorials/custom-ci.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
+---
+title: Custom CI
+toc_max_heading_level: 6
+---
+
+## How does Horizon CI work
+Horizon CI works based on [tekton pipeline](https://github.com/tektoncd/pipeline) and [tekton triggers](https://github.com/tektoncd/triggers). A tekton pipeline will be triggered for a incoming buildDeploy request to perform the typical steps, such as compiling and pushing image. Then, the pipeline output that holds artifact image url and other information will be returned to Horizon core by callback. Horizon CI pipeline includes two tasks that is shown below:
```
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:7]: a incoming  -> an incoming

File: docs/tutorials/custom-ci.md (hunk 4 above)
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:75]: overrode  -> overridden

File: docs/tutorials/custom-ci.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+| cluster          | cluster name                           |                                                              |
+| clusterID        | cluster ID                             |                                                              |
+| environment      | environment name                       |                                                              |
+| gitRepositoryUrl | git repository url of source code      |                                                              |
+| gitSubfolder     | sub folder from context                |                                                              |
+| gitCommit        | git commit of revision                 |                                                              |
+| gitBranch        | git branch of revision                 |                                                              |
+| gitTag           | git tag of revision                    |                                                              |
+| pipelineJSONBlob | json string of build config            | you can parse it by json paser, such as `jq`                 |
```
Comment (iamyeka) [inline docs/tutorials/custom-ci.md:103]: json parser

**Review by iamyeka — CHANGES_REQUESTED** [review]

File: docs/tutorials/custom-ci.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
+---
+title: Custom CI
+toc_max_heading_level: 6
+---
+
+## How does Horizon CI work
+Horizon CI works based on [tekton pipeline](https://github.com/tektoncd/pipeline) and [tekton triggers](https://github.com/tektoncd/triggers). A tekton pipeline will be triggered for a incoming buildDeploy request to perform the typical steps, such as compiling and pushing image. Then, the pipeline output that holds artifact image url and other information will be returned to Horizon core by callback. Horizon CI pipeline includes two tasks that is shown below:
+![pipeline](/image/pipeline.png)
+* Git step: use token to pull user's source code.
```
Comment (work-jlsun) [inline docs/tutorials/custom-ci.md:9]: user's source code. => source code

File: docs/tutorials/custom-ci.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+* Compile step: compile and package the source code, such as `mvn clean package`.
```
Comment (work-jlsun) [inline docs/tutorials/custom-ci.md:10]: add more , such as  `mvn clean packege `, `go build ` `docker build`?

File: docs/tutorials/custom-ci.md (hunk 2 above)
Comment (work-jlsun) [reply docs/tutorials/custom-ci.md:15]: Custom CI is supported partly on Horizon now =>Custom CI is supported,

**Comment by work-jlsun** [comment]
LGTM

**Review by iamyeka — APPROVED** [review]

**Review by iamyeka — APPROVED** [review]
LGTM

**Pull request merged**